### PR TITLE
Added test for uno horizontal_skips_map when None, and make validate_scaling_factor to return None if all scalings are 1 for eff.

### PR DESCRIPTION
**Pull request by btolooshams**

### CODE REVIEW

File: neuralop/layers/complex_utils.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def ctanh(x: torch.Tensor):
+    """Complex-valued tanh stabilizer"""
+    return torch.tanh(x.real).type(torch.cfloat) + 1j * torch.tanh(x.imag).type(
+        torch.cfloat()
+    )
+
+
+def apply_complex(fr, fi, x, dtype=torch.cfloat):
```
Comment (JeanKossaifi) [inline neuralop/layers/complex_utils.py:31]: Could we make the variable names clearer? e.g. expand `f`, use real and imag etc.

File: neuralop/layers/complex_utils.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+class ComplexValued(nn.Module):
+    """
+    Wrapper class that converts a standard nn.Module that operates on real data
+    into a module that operates on complex-valued spatial data.
+    """
+
+    def __init__(self, mod):
+        super(ComplexValued, self).__init__()
+        self.fr = deepcopy(mod)
+        self.fi = deepcopy(mod)
```
Comment (JeanKossaifi) [inline neuralop/layers/complex_utils.py:46]: Is the deep copy needed? Are we expecting the module to be used somewhere else before conversion?

File: neuralop/utils.py [hunk 3]
```diff
@@ -237,14 +237,20 @@ def validate_scaling_factor(
         isinstance(scaling_factor, list)
         and len(scaling_factor) > 0
         and all([isinstance(s, (list)) for s in scaling_factor])
-    ):
+    ):  
+        flag_if_all_one = True # this is to save computation if scaling is 1.
```
Comment (JeanKossaifi) [inline neuralop/utils.py:241]:
```suggestion
        flag_all_ones = True # this is to save computation if scaling is 1.
```

File: neuralop/layers/complex_utils.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    )
+
+
+class ComplexValued(nn.Module):
+    """
+    Wrapper class that converts a standard nn.Module that operates on real data
+    into a module that operates on complex-valued spatial data.
+    """
+
+    def __init__(self, mod):
```
[the remark this answers is not in the document]
Comment (btolooshams) [reply neuralop/layers/complex_utils.py]: updated.

File: neuralop/layers/complex_utils.py (hunk 2 above)
Comment (btolooshams) [reply neuralop/layers/complex_utils.py]: This line of code is from @dhpitt. I am not sure if deepcopy is necessary.

**Comment by btolooshams** [comment]
The latest update here is about complex FNO (main changes are in the spectral convolution). I have also added pytest when the data is complex.

**Comment by JeanKossaifi** [comment]
@btolooshams is this finished? If so shall we revisit and merge? @dhpitt for viz.

**Comment by dhpitt** [comment]
Superseded by #400, closing

**Comment by dhpitt** [comment]
Superseded by #400

**Pull request closed without merging**